examples/knative/python/kn-py-vro/handler.py

```diff
@@ -0,0 +1,270 @@
+            param['type'] = "string"
+            param['value'] = {"string": {"value": value}}
+
+    elif type(value) == bool:
+        param['type'] = "boolean"
+        param['value'] = {"boolean": {"value": value}}
+
+    elif type(value) == dict:  # a dict is likely a VC manged object reference
+        for v in value.values():
+            # search the items of the dict and look for contained dicts with a 'Type' property
+            if type(v) == dict and 'Type' in v:
+                param['type'] = f"VC:{v['Type']}"
+                param['value'] = {
+                    "sdk-object": {
+                        "type": param['type'],
+                        "id": f'{vchost},id:{v["Value"]}'
+                    }
+                }
+        if not 'type' in param: # if no vc managed object was found, pass the data as JSON
+            param['type'] = 'string'
+            param['value'] = {"string": {"value": json.dumps(value)}}
+
+    elif type(value) == list:
+        param['type'] = "Properties"
+        properties = []
+        for v in value:
+            if 'Key' in v and 'Value' in v:
+                # pass the key and value to this function recursively to get a correctly structured value
+                vroParam = getVroInputParam((v["Key"], v["Value"]))
+                properties.append({"key": v["Key"], "value": vroParam["value"]})
+        param['value'] = {"properties": {"property": properties}}
+
+    elif value is None:
+        param = None
+
+    else:  # Unhandled data type - try and wrangle to a string
+        app.logger.debug(f'{bgc.WARNING}Unhandled data type "{type(value).__name__}" - forcing to string{bgc.ENDC}')
+        param['type'] = "string"
+        param['value'] = {"string": {"value": str(value)}}
+
+    if type(param) == dict:  # set the parameter name for valid types
+        param["name"] = name[0].lower() + name[1:]
+    return param
+
+
+def get_response(message, code):
+        app.logger.error(message)
+        msg = {
+            'status': code,
+            'error': message,
+        }
+        resp = jsonify(msg)
+        resp.status_code = code
+        return resp
+
+
+app = Flask(__name__)
+@app.route('/', methods=['POST'])
+def handle():
+    """
+    Handles a request to the function
+
+    """
+    global vchost
+
+    # Load the Events that function gets from vCenter through the Event Router
+    app.logger.debug(f'{bgc.HEADER}Reading Cloud Event: {bgc.ENDC}')
+    cevent = from_http(request.headers, request.get_data(),None)
+
+    if cevent._attributes.get("datacontenttype").lower() != "application/json":
+            return get_response(f'invalid datacontenttype for cloud event: {cevent._attributes.get("datacontenttype")}', 500)
+    app.logger.debug(f'{bgc.OKBLUE}Event > {bgc.ENDC}{type(cevent)} - {cevent}')
+
+    try:
+        # CloudEvent - simple validation
+        source = cevent._attributes['source']
+        event = cevent.data
+        _ = event['Key']
```

Comment:
question: is this needed?

Reply:
I did this as a crude way to confirm the presence of the 'Key' parameter in the event data (i.e. is this valid CloudEvent data and it's all parsed properly, and is a hang up from the OpenFaaS version of the function. I think I'm right in thinking that with Knative and  the CloudEvent datatype, *and* the tests done previous to this that this is no longer serving any useful purpose.

Reply:
Yup, shouldn't be needed.